presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -0,0 +1,222 @@
+                tempDirectory.resolve("cache1"),
+                tempDirectory.resolve("cache2"));
+        for (java.nio.file.Path directory : cacheDirectories) {
+            createDirectories(directory);
+        }
+
+        // initialize rubix in master-only mode
+        RubixConfig rubixConfig = new RubixConfig();
+        rubixConfig.setCacheLocation(join(
+                cacheDirectories.stream()
+                        .map(java.nio.file.Path::toString)
+                        .collect(toImmutableList()),
+                ","));
+        RubixConfigurationInitializer rubixConfigInitializer = new RubixConfigurationInitializer(rubixConfig);
+        configurationInitializer = new HdfsConfigurationInitializer(config, ImmutableSet.of(
+                // make sure that dummy cluster manager is used
+                initConfig -> setPrestoClusterManager(initConfig, DummyClusterManager.class.getName())));
+        RubixInitializer rubixInitializer = new RubixInitializer(
+                new CatalogName("catalog"),
+                rubixConfigInitializer,
+                configurationInitializer);
+        InternalNode coordinatorNode = new InternalNode(
+                "master",
+                URI.create("http://127.0.0.1:8080"),
+                UNKNOWN,
+                true);
+        TestingNodeManager nodeManager = new TestingNodeManager(
+                coordinatorNode,
+                ImmutableList.of());
+        rubixInitializer.initializeRubix(nodeManager);
+
+        // wait for rubix to start
+        Failsafe.with(
+                new RetryPolicy<>()
+                        .withDelay(Duration.ofSeconds(1))
+                        // unlimited attempts
+                        .withMaxAttempts(-1)
+                        .withMaxDuration(Duration.ofMinutes(1)))
+                .run(() -> checkState(rubixConfigInitializer.isCacheReady()));
+
+        // get rubix caching filesystem
+        configurationInitializer = new HdfsConfigurationInitializer(config, ImmutableSet.of(rubixConfigInitializer));
+        configuration = new HiveHdfsConfiguration(configurationInitializer, ImmutableSet.of(
+                (dynamicConfig, ignoredContext, ignoredUri) -> {
+                    // make sure a new FS is created with Rubix caching enabled
+                    setCacheKey(dynamicConfig, "rubix");
+                    dynamicConfig.set("fs.file.impl", CachingLocalFileSystem.class.getName());
+                }));
+        environment = new HdfsEnvironment(configuration, config, new NoHdfsAuthentication());
+
+        cachingFileSystem = environment.getFileSystem(context, path);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws IOException
+    {
+        try (Closer closer = Closer.create()) {
+            closer.register(() -> deleteRecursively(tempDirectory, ALLOW_INSECURE));
+            closer.register(() -> nonCachingFileSystem.close());
```

Comment:
We should expose BookKeeperServer from RubixInitializer and call stopServer() on it in closer. Otherwise it will block bookKeeper port for rest of the tests in the run.

Reply:
I will address that in next commit